File: CItem_proj_2/weapon.py
from citem import CItem
from typing import List

class Weapon(CItem):
    DEFAULT_BAG_SIZE: int = 20
    DEFAULT_QUANTITY: int = 0
    def __init__(self, name: str, bagSize: int = DEFAULT_BAG_SIZE, quantity: int = DEFAULT_QUANTITY):
        super(Weapon,self).__init__(name)
        self._bagSize = bagSize
        if quantity > bagSize:
            self._quantity = self._bagSize
        else:
            self._quantity = quantity
    
    @property
    def bagSize(self) -> int:
        return self._bagSize

    def icreaseBag(self, newBag) -> None:
        self._bagSize = newBag

    @property
    def quantity(self) -> int:
        return self._quantity

    def fire(self) -> None:
        if self._inUse and self._quantity > 0:
            self._quantity -= 1
    
    def refill(self, amount: int) -> None:
        if self._quantity + amount > self._bagSize:
            self._quantity = self._bagSize
        else:
            self._quantity += amount
        

File: CItem_proj_2_cpp/src/Weapon.cpp
#include "Weapon.h"

namespace item {
    const int Weapon::DEFAULT_BAG_SIZE = 20;
    const int Weapon::DEFAULT_QUANTITY = 0;

    Weapon::Weapon(std::string _name, int _bagSize, int _quantity)
        : CItem(_name), bagSize(_bagSize), quantity(_quantity)
    {
        if(quantity > bagSize) quantity = bagSize;
    }

    int Weapon::getBagSize() const{
        return bagSize;
    }
    void Weapon::setBagSize(int newSize) {
        bagSize = newSize;
    }

    int Weapon::getQuantity() const{
        return quantity;
    }
    void Weapon::setQuantity(int newQuant) {
        if(quantity + newQuant > bagSize) quantity = bagSize;
        else quantity += newQuant;
    }

    void Weapon::fire() {
        if(quantity > 0 && getUsing()) quantity--;
    }
}